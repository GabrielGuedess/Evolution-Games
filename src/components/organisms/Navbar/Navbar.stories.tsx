import { Story, Meta } from '@storybook/react/types-6-0';

import { Navbar } from './Navbar';

export default {
  title: 'Organisms/Navbar',
  component: Navbar,
} as Meta;

export const Default: Story = () => <Navbar />;
