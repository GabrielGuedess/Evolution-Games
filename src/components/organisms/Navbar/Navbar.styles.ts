import * as NavigationMenu from '@radix-ui/react-navigation-menu';
import media from 'styled-media-query';

import { Container } from 'components/atoms/Container/Container';
import MediaMatch from 'components/atoms/MediaMatch/MediaMatch';

import styled, { css } from 'styled-components';

export const Wrapper = styled(Container).attrs({ as: 'menu' })`
  ${({ theme }) => css`
    position: absolute;
    inset: 0 0 auto 0;
    padding: 1.4rem 0;
    display: flex;
    align-items: center;
    flex-direction: row;
    z-index: ${theme.layers.menu};
    justify-content: space-between;

    ${media.greaterThan('large')`
      flex-direction: row;
      padding: 3.6rem 0;
    `}
  `}
`;

export const WrapperLogotipo = styled.div`
  order: 2;

  ${media.greaterThan('large')`
    order: initial;
  `}
`;

export const CustomMediaMatch = styled(MediaMatch)`
  order: 3;

  ${media.greaterThan('large')`
    order: initial;
  `}
`;

export const ListTextLink = styled(NavigationMenu.List)`
  display: flex;
  gap: 6.4rem;
`;

export const ItemTextLink = styled(NavigationMenu.Item)`
  list-style: none;
`;

export const LinkTextLink = styled(NavigationMenu.Link)`
  ${({ theme, active }) => css`
    text-decoration: none;
    outline: none;
    font-weight: 500;
    font-size: ${theme.font.sizes.small};
    line-height: 3.2rem;
    text-transform: uppercase;
    transition: color 0.3s linear;

    color: ${active ? theme.colors.primary : theme.colors.whiteText};
    text-shadow: ${active && theme.shadows.text};

    :is(:hover, :focus) {
      color: ${theme.colors.primary};
      text-shadow: ${theme.shadows.text};
    }
  `}
`;

export const WrapperIcons = styled.div`
  display: flex;
  align-items: center;
  gap: 1.5rem;
  order: 1;

  ${media.greaterThan('large')`
    gap: 3rem;
    order: initial;
  `}
`;

export const HeartLink = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.whiteText};
    transition: color 0.3s linear;

    :is(:hover, :focus) {
      cursor: pointer;
      color: ${theme.colors.primary};
    }
  `}
`;
