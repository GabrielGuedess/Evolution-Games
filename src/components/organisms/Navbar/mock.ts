import { GameCartDropProps } from 'components/molecules/GameCartDrop/GameCartDrop';

export const cartItems: GameCartDropProps[] = [
  {
    id: 'the-witcher-3',
    src: '/img/games/theWitcher.jpg',
    genres: ['Action'],
    title: 'The Witcher 3',
    developer: 'CD Projekt',
    quantity: 1,
    platform: 'ps4',
    price: 100,
  },
  {
    id: 'gta-6',
    src: 'https://gmedia.playstation.com/is/image/SIEPDC/grand-theft-auto-5-hero-banner-desktop-01-ps5-en-04mar22?$2400px$',
    genres: ['Action'],
    title: 'GTA 6',
    developer: 'CD Projekt',
    quantity: 1,
    platform: 'ps4',
    price: 100,
  },
  {
    id: 'rainbow-six',
    src: 'https://image.api.playstation.com/vulcan/ap/rnd/202105/1919/6t4GYDWoLSZVeNjAJLTczMg8.jpg?w=1920',
    genres: ['Action'],
    title: 'Rainbow Six',
    developer: 'CD Projekt',
    quantity: 1,
    platform: 'ps4',
    price: 100,
  },
];
