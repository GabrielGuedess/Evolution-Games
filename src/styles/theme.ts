export default {
  grid: {
    container: '165.6rem',
  },
  border: {
    radius: {
      regular: '1rem',
    },
  },
  font: {
    family: {
      primary:
        "'Inter', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    },
    regular: 400,
    medium: 500,
    sizes: {
      xxsmall: '1rem',
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      xmedium: '1.8rem',
      xxmedium: '2.0rem',
      big: '2.2rem',
      xbig: '2.5rem',
      xxbig: '2.8rem',
      xxxbig: '3.0rem',
      large: '3.2rem',
      xlarge: '4.5rem',
      xxlarge: '5.0rem',
      extra: '6.0rem',
    },
  },
  colors: {
    primary: '#7C3AED',
    secondary: '#717B8A',
    mainBg: '#04040C',
    white: '#FFFFFF',
    whiteText: '#F4F4F5',
    gameDetails: '#010914',
    gameGreat: '#0FBF00',
    gameAverage: '#FFF720',
    gameBad: '#FF3333',
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem',
    big: '5.8rem',
    huge: '8.0rem',
    section: '10rem',
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50,
  },
  shadows: {
    text: '0 0.8rem 1rem rgba(118, 74, 241, 0.5)',
    box: '0 0.8rem 2.8rem rgba(118, 74, 241, 0.5)',
  },
} as const;
