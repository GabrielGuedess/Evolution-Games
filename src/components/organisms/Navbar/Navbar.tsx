import Link from 'next/link';
import { useRouter } from 'next/router';

import { useState } from 'react';

import * as NavigationMenu from '@radix-ui/react-navigation-menu';
import * as Popover from '@radix-ui/react-popover';
import { Squash as Hamburger } from 'hamburger-react';
import { Heart } from 'phosphor-react';

import { Logo } from 'components/atoms/Logo/Logo';
import MediaMatch from 'components/atoms/MediaMatch/MediaMatch';
import {
  AvatarUser,
  AvatarUserProps,
} from 'components/molecules/AvatarUser/AvatarUser';
import { CartButton } from 'components/molecules/CartButton/CartButton';
import { MenuMobile } from 'components/molecules/MenuMobile/MenuMobile';

import { CartItems } from './mock';
import * as S from './Navbar.styles';

export type MenuProps = {
  loading?: boolean;
} & AvatarUserProps;

export const Navbar = ({ loading = false, ...props }: MenuProps) => {
  const { asPath } = useRouter();
  const [isOpen, setIsOpen] = useState(false);

  return (
    <S.Wrapper>
      <Logo />

      <MediaMatch greaterThan="medium">
        <NavigationMenu.Root>
          <S.ListTextLink>
            <S.ItemTextLink>
              <Link href="/" passHref>
                <S.LinkTextLink active={asPath === '/'}>Home</S.LinkTextLink>
              </Link>
            </S.ItemTextLink>

            <S.ItemTextLink>
              <Link href="/games" passHref>
                <S.LinkTextLink active={asPath === '/games'}>
                  Explorer
                </S.LinkTextLink>
              </Link>
            </S.ItemTextLink>

            <S.ItemTextLink>
              <Link href="/games" passHref>
                <S.LinkTextLink active={asPath === '/games'}>
                  Contact
                </S.LinkTextLink>
              </Link>
            </S.ItemTextLink>

            <NavigationMenu.Indicator />
          </S.ListTextLink>

          <NavigationMenu.Viewport />
        </NavigationMenu.Root>
      </MediaMatch>

      {!loading && (
        <>
          <S.WrapperIcons>
            <Link href="/favorites" passHref>
              <S.HeartLink aria-label="Link Favorites">
                <Heart size={24} />
              </S.HeartLink>
            </Link>

            <CartButton items={CartItems} />

            <MediaMatch greaterThan="medium">
              <AvatarUser {...props} />
            </MediaMatch>
          </S.WrapperIcons>

          <MediaMatch lessThan="medium">
            <Popover.Root open={isOpen} onOpenChange={setIsOpen}>
              <Popover.Trigger>
                <Hamburger
                  size={23}
                  duration={0.5}
                  label="Show menu"
                  toggled={isOpen}
                  toggle={setIsOpen}
                />
              </Popover.Trigger>
              <Popover.Anchor />
              <MenuMobile path={asPath} {...props} />
            </Popover.Root>
          </MediaMatch>
        </>
      )}
    </S.Wrapper>
  );
};
