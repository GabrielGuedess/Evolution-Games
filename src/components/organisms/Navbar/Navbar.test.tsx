import { renderWithProviders } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import theme from 'styles/theme';

import { Navbar } from './Navbar';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/router'), 'useRouter');

useRouter.mockImplementation(() => ({
  push: jest.fn(),
  query: '',
  asPath: '/',
  route: '/',
}));

describe('<Navbar />', () => {
  it('should render the Navbar correctly', () => {
    // Arrange
    const { container } = renderWithProviders(<Navbar />);

    // Assert
    expect(screen.getByText('Home')).toHaveStyle({
      color: theme.colors.primary,
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
