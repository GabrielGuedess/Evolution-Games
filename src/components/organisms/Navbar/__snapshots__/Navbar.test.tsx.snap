// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Navbar /> should render the Navbar correctly 1'] = `
.c3 {
  width: 8.6rem;
  height: 2rem;
  color: #FFFFFF;
}

.c4 {
  display: none;
}

.c15 {
  display: none;
}

.c13 {
  cursor: pointer;
}

.c14 {
  cursor: pointer;
  color: #F4F4F5;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c14:is(:hover,:focus) {
  color: #7C3AED;
}

.c11 {
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c11:is(:hover,:focus) {
  cursor: pointer;
  color: #7C3AED;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #7C3AED;
  box-shadow: 0 0.8rem 1rem rgba(118,74,241,0.5);
  color: #FFFFFF;
  font-size: 1.2rem;
  line-height: 1rem;
  border-radius: 1rem;
  padding: 0.3rem 0.6rem;
  position: absolute;
  top: -0.4rem;
  right: -0.4rem;
  cursor: pointer;
}

.c0 {
  max-width: 165.6rem;
  margin: 0 1.6rem;
}

.c1 {
  position: absolute;
  inset: 0 0 auto 0;
  padding: 1.4rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  z-index: 20;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  cursor: pointer;
}

.c16 {
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 6.4rem;
}

.c6 {
  list-style: none;
}

.c7 {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  font-weight: 500;
  font-size: 1.4rem;
  line-height: 3.2rem;
  text-transform: uppercase;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
  color: #7C3AED;
  text-shadow: 0 0.8rem 1rem rgba(118,74,241,0.5);
}

.c7:is(:hover,:focus) {
  color: #7C3AED;
  text-shadow: 0 0.8rem 1rem rgba(118,74,241,0.5);
}

.c8 {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  font-weight: 500;
  font-size: 1.4rem;
  line-height: 3.2rem;
  text-transform: uppercase;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
  color: #F4F4F5;
}

.c8:is(:hover,:focus) {
  color: #7C3AED;
  text-shadow: 0 0.8rem 1rem rgba(118,74,241,0.5);
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.5rem;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c10 {
  color: #F4F4F5;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c10:is(:hover,:focus) {
  cursor: pointer;
  color: #7C3AED;
}

@media (min-width:1170px) {
  .c3 {
    width: 13.7rem;
    height: 3.2rem;
    color: #7C3AED;
  }
}

@media (min-width:1170px) {
  .c4 {
    display: block;
  }
}

@media (max-width:1170px) {
  .c15 {
    display: block;
  }
}

@media (min-width:768px) {
  .c12 {
    font-size: 0.8rem;
  }
}

@media (min-width:1170px) {

}

@media (min-width:1170px) {
  .c0 {
    margin: 0 3.2rem;
  }
}

@media (min-width:1440px) {
  .c0 {
    margin: 0 4.8rem;
  }
}

@media (min-width:1656px) {
  .c0 {
    margin-left: auto;
    margin-right: auto;
  }
}

@media (min-width:1170px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 3.6rem 0;
  }
}

@media (min-width:1170px) {
  .c2 {
    -webkit-order: initial;
    -ms-flex-order: initial;
    order: initial;
  }
}

@media (min-width:1170px) {
  .c16 {
    -webkit-order: initial;
    -ms-flex-order: initial;
    order: initial;
  }
}

@media (min-width:1170px) {
  .c9 {
    gap: 3rem;
    -webkit-order: initial;
    -ms-flex-order: initial;
    order: initial;
  }
}

<menu
  class="c0 c1"
>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <svg
        aria-label="Evolution"
        fill="none"
        role="img"
        viewBox="0 0 134 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16.344 18.104V23H0.504V0.919998H16.344V5.816H6.552V9.496H16.28V14.392H6.552V18.104H16.344ZM28.9195 23H24.2795L17.9115 7H24.0875L26.6795 15.704L29.1115 7H35.0955L28.9195 23ZM41.9485 18.36C42.2898 18.936 42.8872 19.224 43.7405 19.224C44.5938 19.224 45.1912 18.936 45.5325 18.36C45.8952 17.784 46.0765 16.696 46.0765 15.096C46.0765 13.4107 45.8952 12.2693 45.5325 11.672C45.1912 11.0747 44.5938 10.776 43.7405 10.776C42.8872 10.776 42.2898 11.0747 41.9485 11.672C41.6072 12.2693 41.4365 13.4107 41.4365 15.096C41.4365 16.696 41.6072 17.784 41.9485 18.36ZM43.7405 6.584C46.5992 6.584 48.6258 7.20267 49.8205 8.44C51.0152 9.656 51.6125 11.8747 51.6125 15.096C51.6125 18.232 51.0045 20.408 49.7885 21.624C48.5938 22.8187 46.5778 23.416 43.7405 23.416C40.9245 23.416 38.9192 22.8187 37.7245 21.624C36.5298 20.408 35.9325 18.232 35.9325 15.096C35.9325 11.8747 36.5298 9.656 37.7245 8.44C38.9192 7.20267 40.9245 6.584 43.7405 6.584Z"
          fill="currentColor"
        />
        <path
          d="M59.5995 0.919998V23H54.0635V0.919998H59.5995ZM67.6888 7V17.464C67.6888 18.1253 67.8381 18.584 68.1368 18.84C68.4568 19.096 68.9368 19.224 69.5768 19.224C70.9208 19.224 71.5928 18.4987 71.5928 17.048V7H77.1287V23H71.5928V21.112H71.0488C70.1314 22.648 68.7981 23.416 67.0488 23.416C65.5768 23.416 64.3928 23.0427 63.4968 22.296C62.6008 21.528 62.1528 20.28 62.1528 18.552V7H67.6888ZM90.5328 18.392V23H86.8208C82.9808 23 81.0608 21.1333 81.0608 17.4V11.32H78.8528V7H81.0608V2.2H86.5968V7H90.5328V11.32H86.5968V16.856C86.5968 17.4107 86.7248 17.8053 86.9808 18.04C87.2368 18.2747 87.6634 18.392 88.2608 18.392H90.5328ZM92.6908 7H98.1948V23H92.6908V7ZM98.3228 5.016H92.5628V0.952H98.3228V5.016ZM106.792 18.36C107.134 18.936 107.731 19.224 108.584 19.224C109.438 19.224 110.035 18.936 110.376 18.36C110.739 17.784 110.92 16.696 110.92 15.096C110.92 13.4107 110.739 12.2693 110.376 11.672C110.035 11.0747 109.438 10.776 108.584 10.776C107.731 10.776 107.134 11.0747 106.792 11.672C106.451 12.2693 106.28 13.4107 106.28 15.096C106.28 16.696 106.451 17.784 106.792 18.36ZM108.584 6.584C111.443 6.584 113.47 7.20267 114.664 8.44C115.859 9.656 116.456 11.8747 116.456 15.096C116.456 18.232 115.848 20.408 114.632 21.624C113.438 22.8187 111.422 23.416 108.584 23.416C105.768 23.416 103.763 22.8187 102.568 21.624C101.374 20.408 100.776 18.232 100.776 15.096C100.776 11.8747 101.374 9.656 102.568 8.44C103.763 7.20267 105.768 6.584 108.584 6.584ZM128.315 23V12.536C128.315 11.8747 128.155 11.416 127.835 11.16C127.537 10.904 127.057 10.776 126.395 10.776C125.051 10.776 124.379 11.5013 124.379 12.952V23H118.875V7H124.379V8.888H124.923C125.819 7.352 127.217 6.584 129.115 6.584C130.651 6.584 131.814 6.95733 132.603 7.704C133.414 8.45067 133.819 9.69867 133.819 11.448V23H128.315Z"
          fill="white"
        />
      </svg>
    </div>
  </div>
  <div
    class="c4"
  >
    <nav
      aria-label="Main"
      data-orientation="horizontal"
      dir="ltr"
    >
      <div
        style="position: relative;"
      >
        <ul
          class="c5"
          data-orientation="horizontal"
          dir="ltr"
        >
          <li
            class="c6"
          >
            <a
              aria-current="page"
              class="c7"
              data-active=""
              data-radix-collection-item=""
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="c6"
          >
            <a
              class="c8"
              data-radix-collection-item=""
              href="/games"
            >
              Explorer
            </a>
          </li>
          <li
            class="c6"
          >
            <a
              class="c8"
              data-radix-collection-item=""
              href="/games"
            >
              Contact
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <div
    class="c9"
  >
    <a
      aria-label="Link Favorites"
      class="c10"
      href="/favorites"
    >
      <svg
        fill="currentColor"
        height="24"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <path
          d="M128,216S28,160,28,92A52,52,0,0,1,128,72h0A52,52,0,0,1,228,92C228,160,128,216,128,216Z"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
      </svg>
    </a>
    <div
      class=""
    >
      <button
        aria-controls="radix-:r4:"
        aria-expanded="false"
        aria-haspopup="dialog"
        data-state="closed"
        type="button"
      >
        <div
          class="c11"
        >
          <svg
            aria-hidden="true"
            aria-label="Cart Icon"
            fill="currentColor"
            focusable="false"
            height="24"
            viewBox="0 0 256 256"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <path
              d="M208.8,72H47.2a8.1,8.1,0,0,0-8,7.1L25,207.1a8,8,0,0,0,7.9,8.9H223.1a8,8,0,0,0,7.9-8.9l-14.2-128A8.1,8.1,0,0,0,208.8,72Z"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
            <path
              d="M88,72a40,40,0,0,1,80,0"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
          </svg>
          <span
            style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
          >
            Shopping Bag
          </span>
          <span
            aria-label="Cart Items"
            class="c12"
            role="img"
          >
            3
          </span>
        </div>
      </button>
      <div />
    </div>
    <div
      class="c4"
    >
      <div
        class=""
      >
        <button
          aria-controls="radix-:r5:"
          aria-expanded="false"
          aria-haspopup="dialog"
          aria-label="User Photo Button"
          class="c13"
          data-state="closed"
          type="button"
        >
          <a
            aria-label="Link for Login"
            class="c14"
            href="/sign-in"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 256 256"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <circle
                cx="128"
                cy="96"
                fill="none"
                r="64"
                stroke="currentColor"
                stroke-miterlimit="10"
                stroke-width="16"
              />
              <path
                d="M31,216a112,112,0,0,1,194,0"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
          </a>
        </button>
        <div />
      </div>
    </div>
  </div>
  <div
    class="c15 c16"
  >
    <button
      aria-controls="radix-:r6:"
      aria-expanded="false"
      aria-haspopup="dialog"
      data-state="closed"
      type="button"
    >
      <div
        aria-expanded="false"
        aria-label="Show menu"
        class="hamburger-react"
        role="button"
        style="cursor: pointer; height: 48px; position: relative; transition: 0.5s cubic-bezier(0, 0, 0, 1); user-select: none; width: 48px; outline: none;"
        tabindex="0"
      >
        <div
          style="transition: 0.25s cubic-bezier(0, 0, 0, 1) 0.25s; transform: none;"
        >
          <div
            style="background: currentColor; height: 2px; left: 13px; position: absolute; width: 23px; top: 16px; transition: 0.25s cubic-bezier(0, 0, 0, 1) 0s; transform: none;"
          />
        </div>
        <div
          style="transition: 0.25s cubic-bezier(0, 0, 0, 1); opacity: 1;"
        >
          <div
            style="background: currentColor; height: 2px; left: 13px; position: absolute; width: 23px; top: 23px; transition: 0.25s cubic-bezier(0, 0, 0, 1);"
          />
        </div>
        <div
          style="transition: 0.25s cubic-bezier(0, 0, 0, 1) 0.25s; transform: none;"
        >
          <div
            style="background: currentColor; height: 2px; left: 13px; position: absolute; width: 23px; top: 30px; transition: 0.25s cubic-bezier(0, 0, 0, 1) 0s; transform: none;"
          />
        </div>
      </div>
    </button>
    <div />
  </div>
</menu>
`;
